### PR TITLE
Optimise project structure

### PR DESCRIPTION
**Pull request by ivanterekh**
- Moved everething to cmd and pkg folders.
- Put all user functionality to user package.

### CODE REVIEW

File: pkg/auth/authenticator.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ type Authenticator struct {
 	exp    time.Duration
 	method *jwt.SigningMethodHMAC
 	secret string
-	users  users.UserRepository
+	users  repository.UserRepository
```
Comment (dzyanis) [inline pkg/auth/authenticator.go]: In this case,  you can call it just `users.Repository`

File: Makefile [hunk 2]
```diff
@@ -13,7 +13,8 @@ build:
 		-X ${MODULE}/version.Version=${VERSION} \
 		-X ${MODULE}/version.Commit=$(shell git rev-parse HEAD) \
 		-X ${MODULE}/version.BuildTime=$(shell date -u '+%Y-%m-%d_%H:%M:%S')" \
-		-o ${APP}
+		-o ${APP} \
+		./cmd/go-skeleton
```
Comment (dzyanis) [inline Makefile:17]: makefile doesn't work for me. Add information about requirements and how to  install the app

File: pkg/auth/authenticator.go [hunk 3]
```diff
@@ -69,7 +69,7 @@ func (a *Authenticator) GenToken(email, password string) (string, error) {
 }
 
 // Authenticate returns user if token is valid.
-func (a *Authenticator) Authenticate(tokenStr string) (*model.User, error) {
+func (a *Authenticator) Authenticate(tokenStr string) (*user.User, error) {
```
Comment (dzyanis) [inline pkg/auth/authenticator.go]: I'd prefer `users.User` https://blog.golang.org/package-names

File: pkg/user/repository/repository.go [hunk 4]
```diff
@@ -1,9 +1,9 @@
-package users
+package repository
```
Comment (dzyanis) [inline pkg/user/repository/repository.go]: I think right now there is no real need to have so many packages. Package `users` and there are files model, repository and etc.

**Review by dzyanis — CHANGES_REQUESTED** [review]

**Comment by obalunenko** [comment]
What is the purpose of moving all packages to  `/pkg`?
All that located in pkg is expected to be library code that's ok to use by external applications (e.g., /pkg/mypubliclib). Other projects will import these libraries expecting them to work, so think twice before you put something here 
https://github.com/golang-standards/project-layout/blob/master/pkg/README.md

Suggest to move your packages to `internal` as they would not be exported to another projects or leave them in the root of repo
Detailed description of projects layout you can find here:
https://github.com/golang-standards/project-layout/

**Pull request merged**